test/EcdsaXi2023Cryptosuite.spec.js

```diff
@@ -0,0 +1,379 @@
+/*!
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+import {expect} from 'chai';
+
+import jsigs from 'jsonld-signatures';
+const {purposes: {AssertionProofPurpose}} = jsigs;
+
+import * as EcdsaMultikey from '@digitalbazaar/ecdsa-multikey';
+import {
+  credential,
+  ecdsaMultikeyKeyPair,
+  ecdsaSecp256KeyPair
+} from './mock-data.js';
+import {createCryptosuite} from '../lib/index.js';
+import {DataIntegrityProof} from '@digitalbazaar/data-integrity';
+
+import {loader} from './documentLoader.js';
+
+const documentLoader = loader.build();
+const extraInformation = new Uint8Array([12, 52, 75, 63, 74, 85, 21, 5, 62, 10,
+  12, 52, 75, 63, 74, 85, 21, 5, 62, 100,
+  12, 52, 75, 63, 74, 85, 21, 5, 62, 100,
+  12, 52, 75, 63, 74, 85, 21, 5, 62, 100,
+  12, 52, 75, 63, 74, 85, 21, 5, 62, 100,
+  12, 52, 75, 63, 74, 85, 21, 5, 62, 100,
+  12, 52, 75, 63
+]);
+const ecdsaXi2023Cryptosuite = createCryptosuite({extraInformation});
+
+describe('EcdsaXi2023Cryptosuite', () => {
+  describe('exports', () => {
+    it('it should have proper exports', async () => {
+      should.exist(ecdsaXi2023Cryptosuite);
+      ecdsaXi2023Cryptosuite.name.should.equal('ecdsa-xi-2023');
+      ecdsaXi2023Cryptosuite.requiredAlgorithm.should.eql(['P-256', 'P-384']);
+      ecdsaXi2023Cryptosuite.canonize.should.be.a('function');
+      ecdsaXi2023Cryptosuite.createVerifier.should.be.a('function');
+    });
+  });
+
+  describe('canonize()', () => {
+    it('should canonize using URDNA2015 w/ n-quads', async () => {
+      const unsignedCredential = JSON.parse(JSON.stringify(credential));
+
+      let result;
+      let error;
+      try {
+        result = await ecdsaXi2023Cryptosuite.canonize(
+          unsignedCredential, {documentLoader});
+      } catch(e) {
+        error = e;
+      }
+
+      expect(error).to.not.exist;
+      expect(result).to.exist;
+      /* eslint-disable max-len */
+      const expectedResult = `<http://example.edu/credentials/1872> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <https://schema.org#AlumniCredential> .
```

Comment:
Should change the `http` => `https` and regen stuff. It's not technically wrong to use an `http` URL in a VC like this, but as it could cause some security-related problem, we should expect someone to complain about it at some point and we should just fix it up while we're here.